src/Element/BaseBlock.php

```
@@ -65,20 +75,94 @@ abstract class BaseBlock extends BaseTag
+            throw new LogicException(
+                Message::UNEXPECTED_END_CALL_NO_BEGIN->getMessage(static::class),
+            );
+        }
+
+        $tag = array_pop($stack);
+
+        if ($stack === []) {
+            self::$stack?->offsetUnset($key);
+        } else {
+            self::$stack?->offsetSet($key, $stack);
+        }
```

PHPMD (2.15.0)

[error] The method end uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)